library/init/meta/interactive.lean

```diff
@@ -6,8 +6,11 @@ Authors: Leonardo de Moura
 prelude
 import init.meta.tactic init.meta.rewrite_tactic init.meta.simp_tactic
 import init.meta.smt.congruence_closure init.category.combinators
+import init.meta.lean_parser
 
 namespace interactive
+@[reducible] meta def parse {α : Type} (p : lean_parser α) : Type := α
```

Comment:
> I wrote some functions for some types by hand, but I don't see how to make it general enough.

I think that's okay, we can just introduce a typeclass for that. It would not be any less general or more work than outright returning `pexpr` from the parameter parsers.

```
meta class has_quote (α : Type) :=
(quote : α -> pexpr)

@[reducible] meta def parse {α : Type} [has_quote α] (p : lean_parser pexpr) : Type := α
```